enigma-swing/src/main/java/cuchaz/enigma/gui/search/SearchUtil.java

```diff
@@ -134,7 +136,10 @@ public float getScore(String term, int hits) {
 			float maxScore = (float) Arrays.stream(this.components)
 					.mapToDouble(name -> getScoreFor(ucTerm, name))
 					.max().orElse(0.0);
-			return maxScore * (hits + 1);
+
+			// modify by type
+			int typeModifier = ((Math.abs(this.searchEntry.getType().ordinal() - SearchDialog.Type.values().length)) + 1);
```

Comment:
The `abs` isn't needed since `ordinal()` is always less than `values().length`. If the goal of this is to put some entries before others I don't think the `+ 1` is needed either
```suggestion
			int typeModifier = SearchDialog.Type.values().length - this.searchEntry.getType().ordinal() + 1;
```

Reply:
the +1 was a safety check to make sure the modifier never resolved to zero